samples/oauth_rails_vethealth/app/models/health_api_response.rb

```diff
@@ -0,0 +1,46 @@
+class HealthApiResponse
+  attr_reader :api_name, :target, :page, :count
+  def initialize(api_name:, id:, access_token:, action: :read, page: 1, count: 10)
+    @api_name, @id, @action = api_name, id, action
+    if action == :search
+      @page, @count = page, count
+      search_param = (api_name == 'Patient' ? '_id' : 'patient')
+      @target = "https://dev-api.va.gov/services/argonaut/v0/#{api_name}?#{search_param}=#{id}&page=#{page}&_count=#{count}"
+    elsif action == :read
+      @page, @count = nil, nil
+      @target = "https://dev-api.va.gov/services/argonaut/v0/#{api_name}/#{id}"
+    else
+      raise "action must be one of the symbols :search or :read"
+    end
+    @api_response = HTTParty.get(@target, headers: { Authorization: "Bearer #{access_token}" })
+  end
+
+  def code
+    @api_response.code
+  end
+
+  def response_string
+    return @response_string if @response_string
```

Comment:
How is `@response_string` originally being set? I don't see it anywhere else

Reply:
I tend to memoize complex methods so the logic doesn't run every time a method is called.  This method was really complex when I added it.  It's still doing a lot of string substitution and stuff, so I figured I'd leave it in.

Reply:
In that case, this entire method can be one line:
`@response_string ||= JSON.pretty_generate(@api_response.to_h).gsub(/\"(https:\/\/dev-api.va.gov\/services\/argonaut\/v0)\/(\w+)\/(.+)\"/, "<a href=\"/health_api/api_response/\\2/\\3\">\\1\/\\2\/\\3</a>")`